src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2793,30 +3038,9 @@ var projectUniqueNamesForBuildIntToUpdate
                 var now = DateTime.UtcNow;
                 void cacheContextModifier(SourceCacheContext c) => c.MaxAge = now;
 
-                // Check if current project is there in update cache and needs revaluation
-                var isProjectUpdated = false;
-                if (_buildIntegratedProjectsUpdateDict != null &&
-                    _buildIntegratedProjectsUpdateDict.TryGetValue(
-                        buildIntegratedProject.MSBuildProjectPath,
-                        out isProjectUpdated) &&
-                    isProjectUpdated)
-                {
-                    await DependencyGraphRestoreUtility.RestoreProjectAsync(
-                            SolutionManager,
-                            buildIntegratedProject,
-                            referenceContext,
-                            GetRestoreProviderCache(),
-                            cacheContextModifier,
-                            projectAction.Sources,
-                            nuGetProjectContext.OperationId,
-                            logger,
-                            token);
-                }
-                else
-                {
-                    // Write out the lock file
-                    await RestoreRunner.CommitAsync(projectAction.RestoreResultPair, token);
-                }
+                // Write out the lock file, now no need bubbling re-evaluating of parent projects when you restore from PM UI.
+                // We already taken account of that concern in PreviewBuildIntegratedProjectsActionsAsync method.
+                await RestoreRunner.CommitAsync(projectAction.RestoreResultPair, token);
```

Comment:
Here I manually created above scenario:
Parent1 = `ConsoleApp2`
Parent2 = `CoParentConsoleApp3`
Child1 = `ConsoleApp1`
Child 2= `MainApplication.Xplat`

![image](https://user-images.githubusercontent.com/8766776/91007545-1bf0b580-e591-11ea-82b5-05a69579a851.png)

Here is [log for update all projects](https://gist.github.com/erdembayar/8ba2dcde21ac156cd890e8ddf5fb8db7).
Here is [log for only update Child1 & Child2](https://gist.github.com/erdembayar/9e3dbd3322fec60fb86cd8f6444a13ed).
Here is [log for only update Parent1](https://gist.github.com/erdembayar/67614d9ae7889f8c9777af8e1b9b31bc).
Here is [log for only update Parent2](https://gist.github.com/erdembayar/6ca19b800329bb23504b79dd52639abe).

None of above cases all project including parent projects are restored only once.

Reply:
The 2nd example https://gist.github.com/erdembayar/9e3dbd3322fec60fb86cd8f6444a13ed shows multiple restores for CoParentConsoleApp3.

In particular 

> Assets file has not changed. Skipping assets file writing. Path: C:\Users\eryondon\Downloads\MainApplication.Xplat\MainApplication.Xplat\CoParentConsoleApp3\obj\project.assets.json

I don't know why the messages such as `Restoring` etc are not showing up there, but the code suggested that it happened multiple times so I dug into the log message to try to explain it. 

Can you please take a look?

Reply:
@nkolev92 
I committed new change which remembers restored parents. So now check below log:
https://gist.github.com/erdembayar/18c19ad954749e7966b32b7c3559ae61

Only question is maybe instead of adding to `_buildIntegratedProjectsUpdateSet` set I create another set with also tracks packageId which caused evaluation of parent for projects. So when different package from another child causes evaluation then we still need to evaluate again.

Reply:
logs look better. 

> Only question is maybe instead of adding to _buildIntegratedProjectsUpdateSet set I create another set with also tracks packageId which caused evaluation of parent for projects. So when different package from another child causes evaluation then we still need to evaluate again.

Why does the package id matter?

Think of the potential changes to the project as one changeset. 

You apply *all* changes at once...then you restore *all* parents. 
It's the complete operation status that matters.

Reply:
Ok. We'll discuss tomorrow one more time during 1:1.

Reply:
Addressed your concern now it's moved to 1 line in Line 3116.